sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -684,13 +614,14 @@ export class EventHubSender extends LinkEntity {
 
           try {
             await defaultLock.acquire(this.senderLock, () => {
-              return this._init();
+              return this._init(
+                this._createSenderOptions({
+                  sendTimeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions)
```

Comment:
`getRetryAttemptTimeoutInMs` returns time in milliseconds, whereas `sendTimeoutInSeconds` expects this to be in seconds.

I would suggest to refactor `this._createSenderOptions` function as below
- Take `timeoutInMs` as mandatory parameter
- Take `newName` as optional parameter
- Delete the interface `CreateSenderOptions`
- `_createSenderOptions` does the conversion to seconds before returning the "options", so that we pass in the right unit of time to rhea-promise

Reply:
Updated, with these `init` needed options to be made mandatory since `timeoutInMs` is now mandatory.